shard-operator-parent/shard-operator-core/src/main/java/com/redhat/service/smartevents/shard/operator/core/resources/CustomResourceStatus.java

```diff
@@ -116,4 +122,8 @@ public void markConditionFailed(final String conditionType) {
         markConditionFailed(conditionType, null, "", null);
     }
 
+    public void removeCondition(String conditionType) {
```

Comment:
I don't think we should EVER remove a `Condition`.

If, for whatever reason, the `Condition` is already `True` (IDK, perhaps a shared Kafka Source/Sink?) then the `Condition` should simply be marked as `READY`.